extensions/amp-list/validator-amp-list.protoascii

```diff
@@ -94,6 +105,33 @@ tags: {  # <amp-list> with mandatory src and/or [src] attr
     supported_layouts: RESPONSIVE
   }
 }
+tags: { # amp-list load-more children
```

Comment:
Basically `amp-list` with `load-more` should be allowed to have up to 4 div children, each containing one of these attributes. These attributes should be mutually exclusive, and there should only be at most one div containing each attribute. If there's a better way to "validate" that, I'd love to know. Otherwise, I'll just write some JS asserts in the source.

Reply:
I think the alternative here is to define a reference point for `amp list [load-more]` and have that be a mandatory parent, but I'm not sure where we are on the usage of reference points discussion (the last I heard we are refactoring / changing the way we use them?).

Reply:
Actually, per our conversation in https://github.com/ampproject/amphtml/pull/19967, it's highly likely that we would like to allow this to be any element, not just a `<div>`. Is there better way to validate this without whitelisting all the allowed tags here, or introducing reference points?

Reply:
You could add these attributes to $GLOBALS and they would be allowed on any element. Then `requires_extension` could be added to them to ensure that `AMP-LIST` is included as a script tag. However, the requirement of having the parent be `amp-list` is then lost as that's not possible to put in an attribute spec since it really only applies to tags. Further they could then also be put on the `<amp-list>` element and I'm not sure what that would even mean.

Reply:
If it does move to being on any element I'd suggest making the attribute name more explicit about amp-list, e.g. amp-list-load-more-button.

Reply:
Another approach: create a new tag spec amp-list-load-more and it can have one of these attributes (button, failed, end, loading). Then mandatory_parent can be enforced as well and the developer can put whatever elements they want inside.

Reply:
Updated the code per your most recent suggestion. Does that look right to you? I removed the `mandatory_oneof` because `<amp-list>` will also contain default children. It always contains a `<template>` element, and at least currently, allows any other arbitrary children and we can't break backwards compat.

Reply:
That wasn't my suggestion. I suggested making a new tag spec with the tag being amp-list-load-more.

Reply:
Apologies I misunderstood. Allow me to clarify: as in literally creating a new tag called `<amp-list-load-more>`?

Reply:
Yes.

Reply:
Awesome. That does seem like the ideal solution. Especially since that will also  let us cleanly whitelist which children we want to allow in these elements. Thanks a lot @honeybadgerdontcare for the offline followup as well!

Reply:
This is done.